### DISCUSSION TITLE
CLI: Aborting the singleFile html save process from browser script

### DISCUSSION THREAD

**Discussion by mayursarojdev** [gildas-lormeau/SingleFile#1086] [Q&A]
Would really appreciate if you could provide pointers on how passed browser script running in the singleFile browser instance depending upon some conditions it will let the browser continue saving single file html else abort and throw custom error.
In our case we are checking for redirected url and also dom elements and in certain cases we dont need to save the single file.

Thank you

**Comment by gildas-lormeau** [comment]
This example of code should work.
```js
addEventListener("single-file-on-before-capture-request", event => {
  if (cancelSave) {
    event.preventDefault();
    throw new Error("Save cancelled");
  }
});
```
